### PR TITLE
feat: add apache pegasus store

### PR DESCRIPTION
**Pull request by feiguoL**
[**Pegasus]**(https://github.com/apache/incubator-pegasus) targets to fill the gap between Redis and HBase, it is a distributed Key-Value storage system. it's design goal:

- **High scalability** : Distributed horizontal expansion is achieved through hash sharding.
- **Strong consistency** : Guaranteed by PacificA consistency agreement.
- **High performance** : The bottom layer uses RocksDB as the storage engine.
- **Simple and easy to use** : a good interface based on Key-Value.

benchmark data reference [**document**](https://pegasus.apache.org/overview/benchmark/)

use local pegasus onebox test result:
image
image

### CODE REVIEW

File: store/pegasus.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+	"github.com/spf13/cast"
+	"strings"
+	"time"
+)
+
+const (
+	// PegasusType represents the storage type as a string value
+	PegasusType = "pegasus"
+	// PegasusTagPattern represents the tag pattern to be used as a key in specified storage
+	PegasusTagPattern = "go_cache_tag_%s"
```
Comment (eko) [inline store/pegasus.go:18]: Could you please use « gocache_tag_%s » please? Same for DefaultTable below.

File: store/pegasus.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+}
+
+// Close when exit store
+func (p *PegasusStore) Close() error {
+	return p.client.Close()
+}
+
+// Get returns data stored from a given key
+func (p *PegasusStore) Get(key interface{}) (interface{}, error) {
+	table, err := p.client.OpenTable(context.Background(), p.options.TableName)
```
Comment (eko) [inline store/pegasus.go:127]: Note to myself: I will pass parent context to these methods directly

**Review by eko — COMMENTED** [review]
Thanks for adding Pegasus store support!

Could you please fix the little wording issue in tag key name and table name so we could merge?

Thank you!

File: store/pegasus.go (hunk 1 above)
Comment (feiguoL) [reply store/pegasus.go:18]: ok, i will do it

File: store/pegasus.go (hunk 2 above)
Comment (feiguoL) [reply store/pegasus.go]: done!

**Comment by eko** [comment]
Perfect, thank you @feiguoL!

**Pull request merged**